esmvaltool/diag_scripts/weighting/weighted_temperature_graph.py

```diff
@@ -0,0 +1,204 @@
+"""Implementation of the climwip weighting scheme.
+
+Lukas Brunner et al. section 2.4
+https://iopscience.iop.org/article/10.1088/1748-9326/ab492f
+"""
+import logging
+import os
+from pathlib import Path
+
+import yaml
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from climwip import log_provenance, read_metadata, read_model_data
+
+from esmvaltool.diag_scripts.shared import get_plot_filename, run_diagnostic
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def read_weights(filename: str):
+    """Read a `.yml` file into a Pandas Dataframe."""
+    with open(filename, 'r') as file:
+        weights = yaml.safe_load(file)
+    return weights
+
+
+def weighted_quantile(values,
+                      quantiles,
+                      sample_weight=None,
+                      values_sorted=False,
+                      old_style=False):
+    """Very close to numpy.percentile, but supports weights.
+
+    source: https://stackoverflow.com/a/29677616/6012085
+
+    NOTE: quantiles should be in [0, 1]!
+    :param values: numpy.array with data
+    :param quantiles: array-like with many quantiles needed
+    :param sample_weight: array-like of the same length as `array`
+    :param values_sorted: bool, if True, then will avoid sorting of
+        initial array
+    :param old_style: if True, will correct output to be consistent
+        with numpy.percentile.
+    :return: numpy.array with computed quantiles.
```

Comment:
If you're going to write docstrings, please use [numpy style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)